### ISSUE TITLE
Error: Sorry, your asciicast is too big.

### ISSUE THREAD

**Issue by drvinceknight** [asciinema/asciinema#91]
I can't seem to upload any casts?

Have just done one that was very short in time, is the size dependent on the amount of text being displayed? (I am for example using a pip install so perhaps that's the problem).

**Comment by daixtr** [comment]
I had this error too..i save locally instead and saw its only 4mb
On Apr 10, 2015 4:22 PM, "vince" <notifications@github.com> wrote:
>
> I can't seem to upload any casts?
>
> Have just done one that was very short in time, is the size dependent on
the amount of text being displayed? (I am for example using a pip install
so perhaps that's the problem).

**Comment by drvinceknight** [comment]
Using `pip install -q` to greatly decrease the output allowed me to upload. Perhaps something about the size limit could be added to the documentation? (Happy to work on a PR if this is a welcome addition)

**Comment by sickill** [comment]
The max size was set 2MB which appears to be too low. I have upped it to 5MB. This isn't much but I'm paying for the storage (S3) from my own pocket so I can't offer GBs of storage for every user. Let me know if that works for you. I'm fine with increasing it even more, but now I want to figure out the good middle ground between user needs and hosting costs.

**Comment by drvinceknight** [comment]
That's cool @sickill , do you have a monetization plan? Perhaps people could pay to lift the limit?

**Comment by drvinceknight** [comment]
How would you feel about something about the limit being in the documentation? (I missed it if it's already there)

**Comment by sickill** [comment]
Paying for lifting the limit sounds interesting, yeah.
I'll add info about limit to docs. Thanks!

**Comment by sickill** [comment]
@drvinceknight @daixtr is the current limit good enough for now? Still hitting the limit?

**Comment by drvinceknight** [comment]
Nope, current limit works great for me. Thanks!

**Comment by morissette** [comment]
:( 

[mharris@mori example]$ asciinema upload /tmp/tmp0p8dkaof-asciinema.json
~ Upload failed: Sorry, your asciicast is too big.
~ Retry later by running: asciinema upload /tmp/tmp0p8dkaof-asciinema.json
[mharris@mori example]$ ls -lah /tmp/tmp0p8dkaof-asciinema.json
-rw------- 1 mharris mharris 8.5M Apr 12 20:51 /tmp/tmp0p8dkaof-asciinema.json

**Comment by caglar10ur** [comment]
How about storing those files gzip'ed in S3? I just produces 21M file and realized that there is a limit but compressing the same file ends up with less than 1M file which I believe less than the current limit.

**Comment by sickill** [comment]
@caglar10ur asciinema-server actually stores the recordings gzipped on S3 :)

The thing is, if we would allow gzipped uploads from the client to the server, then 5M gzipped file would actually be 100M ungzipped when later downloaded by the browser (files on S3 are served with `Transfer-Encoding: gzip` so they're automatically ungzipped by the browser before handed to XHR result callback). 

From this 100M JSON the player builds frames, transforming each stdout event into full screen contents (lines with groups of colored text etc). This whole thing probably takes 10x the memory the fetched JSON takes.

In other words, my concern here is high memory usage in the browser.

One option to optimize mem usage would be to fetch the recording in gzipped form, without `Transfer-Encoding: gzip` header, and ungzipping it on the go, in lazy/streaming fashion. This way we're losing super-fast native ungzipping provided by the browser, and I am yet to find a streaming/incremental gzip decoder for JS (I'd love to be pointed to one if such exists).

I'm open for further discussion of this!

**Comment by massimolepore** [comment]
@sickill Running into "~ Upload failed: Sorry, your asciicast is too big." with a 1.1M file, which is now the current limit for files?

**Comment by sickill** [comment]
@massimolepore no, that's not to supposed to have changed. I think I may have made a mistake when reconfiguring the load balancer few days ago. Let me check that.

**Comment by sickill** [comment]
Fixed, it's 5M again. Sorry about that!

**Comment by sickill** [comment]
FYI the limit is 8 MB now (for a while now actually).

**Comment by pscriptos** [comment]
Hello,

do I have the possibility to set the maximum upload size on a self-hosted instance?

I am using asciinema in a docker and had a longer session tonight which caused my file to grow about 13 MB in the end.
Unfortunately, I can't upload it because I get the following error:


```
➜  ~ asciinema upload pix
asciinema: upload failed: Sorry, your asciicast is too big.
asciinema: retry later by running: asciinema upload pix
```

`-rw-r--r-- 1 root root  13M Feb  2 22:46 pix`

Thank you very much for your great work!

**Comment by ku1ik** [comment]
Hey @pscriptos!

Right now you'd need to customize the image as shown here: https://docs.asciinema.org/manual/server/self-hosting/customization/

But having an easy option (e.g. env var) is something that would useful. I'll see if I can implement that quickly.

**Comment by pscriptos** [comment]
That would be fantastic. Thank you very much! :)

**Comment by ku1ik** [comment]
@pscriptos good news. I've just released version 20240203, with new `UPLOAD_SIZE_LIMIT` option. See here: https://docs.asciinema.org/manual/server/self-hosting/configuration/#upload-size-limit

**Comment by pscriptos** [comment]
You are my personal hero. Thank you so much! 🍻
